kornia/augmentation/augmentation.py

```diff
@@ -2159,7 +2159,11 @@ def __repr__(self) -> str:
 
     def generate_parameters(self, shape: torch.Size) -> Dict[str, torch.Tensor]:
         B, _, H, W = shape
-        return dict(noise=torch.rand(B, 2, H, W) * 2 - 1)
+        if self.same_on_batch:
+            noise = torch.rand(1, 2, H, W).repeat(B, 1, 1, 1)
+        else:
+            noise = torch.rand(B, 2, H, W)
+        return dict(noise=noise * 2 - 1)
```

Comment:
This controls where the random numbers will be generated. The GPU and CPU random states are different. We  provided a control like this:
```python
torch.rand(1, 2, H, W, device=self.device, dtype=self.dtype)
```